conda/plugins/types.py

```diff
@@ -102,3 +104,30 @@ class CondaPostCommand(NamedTuple):
     name: str
     action: Callable[[str], None]
     run_for: set[str]
+
+
+class ChannelAuthBase(AuthBase):
+    """
+    Base class that we require all plugin implementations to use to be compatible.
+
+    Authentication is tightly coupled with individual channel; therefore, an additional
+    ``channel_name`` property must be set on the ``requests.auth.AuthBase`` class.
+    """
+
+    def __init__(self, channel_name: str):
+        self.channel_name = channel_name
+        super().__init__()
```

Comment:
Will this make it harder to re-use existing requests auth handler classes? I would suggest to provide a mixin class for those cases.

```suggestion
class ChannelAuthMixin:
    """
    Class mixin to make all plugin implementations that use an existing
    (e.g. 3rd party) requests authentication handler compatible.
    
    Please use the concrete :class:`conda.plugins.types.ChannelAuthBase`
    in case you're creating an own implementation.
    """

    def __init__(self, channel_name: str):
        self.channel_name = channel_name
        super().__init__()


class ChannelAuthBase(ChannelAuthMixin, AuthBase):
    """
    Base class that we require all plugin implementations to use to be compatible.

    Authentication is tightly coupled with individual channel; therefore, an additional
    ``channel_name`` property must be set on the ``requests.auth.AuthBase`` class.
    """
```

Reply:
I thought about following the mixin pattern here first, but then realized we would have two classes both wanting to define their own `__init__` method. That won't work, right? 😅

Reply:
Yeah, indeed, the method resolution order matters, so using `super()` is needed and following the right order. But since this is an API that would be used not rarely and the authors will likely be advanced enough IMO, I think it's not a problem to expect that.